### PR TITLE
Fix gamfiy bugs

### LINKED ISSUE #654: Gamify does not count subtasks that were marked as done by using the checkbox inside parent task
Description and reproduction:

Create a task with one or more subtasks. Tick the checkbox inside the parent tasks description. The task is marked as done, however the counter is not increased.

If the subtasks are marked as done via right-click or Ctrl-D, it works.

### LINKED ISSUE #655: Gamify does not count subtasks when closing the parent task
Description and reproduction:

Create a task with with n different subtasks. Mark the parent task as done with n open subtasks. The counter is increased by one instead of n (or n+1 don't know how this is counted)

### PR DESCRIPTION
**Pull request by zeddo123**
Fixes #654 
Fixes #655

### CODE REVIEW

File: GTG/core/requester.py [hunk 1]
```diff
@@ -36,6 +36,10 @@ class Requester(GObject.GObject):
     Multiple L{Requester}s can exist on the same datastore, so they should
     never have state of their own.
     """
+    __string_signal__ = (GObject.SignalFlags.RUN_FIRST, None, (str, str,))
+    __none_signal__ = (GObject.SignalFlags.RUN_FIRST, None, tuple())
```
Comment (Neui) [inline GTG/core/requester.py:40]: Never used?

File: GTG/core/requester.py [hunk 2]
```diff
@@ -36,6 +36,10 @@ class Requester(GObject.GObject):
     Multiple L{Requester}s can exist on the same datastore, so they should
     never have state of their own.
     """
+    __string_signal__ = (GObject.SignalFlags.RUN_FIRST, None, (str, str,))
```
Comment (Neui) [inline GTG/core/requester.py:39]: Why not inline it below?

File: GTG/plugins/gamify/gamify.py [hunk 3]
```diff
@@ -190,7 +188,13 @@ def get_number_of_tasks(self):
     def get_streak(self):
         return self.data['streak']
 
-    def on_marked_as_done(self, sender, task_id):
+    def on_status_changed(self, sender, task_id, status):
+        if status == "Done":
```
Comment (Neui) [inline GTG/plugins/gamify/gamify.py:192]: I'd use `STA_DONE` since this is what other code uses

**Review by Neui — COMMENTED** [review]
Not tested the change, but the modifications look ok

File: GTG/core/requester.py (hunk 2 above)
Comment (Neui) [reply GTG/core/requester.py:39]: I mean something like

```python3
    __gsignals__ = {'status-changed': (GObject.SignalFlags.RUN_FIRST, None, (str, str,))}
```

**Comment by diegogangl** [comment]
LGTM, thanks! (and thanks @Neui for the review)

**Pull request merged**